### PR TITLE
Dacheson/add validation checks to model instantiator

### PR DESCRIPTION
**Pull request by drew-acheson**

### CODE REVIEW

File: src/Models/ModelInstantiatorTest.php [hunk 1]
```diff
@@ -21,17 +22,24 @@ public function testArrayToDataObject(): void
             'firstName'    => 'Peter',
 
             // testProperty exists in the class but does not have a Column attribute.
-            'testProperty' => 'test',
+            'testProperty' => true,
```
Comment (pcrist) [inline src/Models/ModelInstantiatorTest.php:25]: Why the change here?

File: src/Models/ModelInstantiatorTest.php [hunk 2]
```diff
@@ -21,17 +22,24 @@ public function testArrayToDataObject(): void
...
             // fakeProperty does not exist in the class.
             'fakeProperty' => 'faker',
         ];
 
-        // Convert the array to a data object.
-        $data_object = $this->model_instantiator->arrayToDataObject(
-            $array,
-            ModelInstantiatorTestObject::class
-        );
+        try
```
Comment (pcrist) [inline src/Models/ModelInstantiatorTest.php:31]: Is the try/catch necessary? We don't expect the function to throw an exception, right?

**Review by pcrist — APPROVED** [review]

File: src/Models/ModelInstantiatorTest.php (hunk 1 above)
Comment (drew-acheson) [reply src/Models/ModelInstantiatorTest.php:25]: Because the property is defined as a bool in the class which caused the validation to fail when it's set as a string here.

File: src/Models/ModelInstantiatorTest.php (hunk 2 above)
Comment (pcrist) [reply src/Models/ModelInstantiatorTest.php:31]: Works for me.

File: src/Models/ModelInstantiatorTest.php (hunk 1 above)
Comment (pcrist) [reply src/Models/ModelInstantiatorTest.php:25]: All right. Apologies if we already discussed this; do we want to validate properties that don't have `Column` attributes?

Comment (drew-acheson) [reply src/Models/ModelInstantiatorTest.php:25]: I'd say yes, since the function is converting an array to a data object based on a class definition and the attribute is defined on the class. This specific function doesn't seem to care about `Column` just attributes.

**Review by bbotto-pdga — APPROVED** [review]

File: src/Models/ModelInstantiatorTest.php (hunk 1 above)
Comment (pcrist) [reply src/Models/ModelInstantiatorTest.php:25]: `arrayToDataObject` ignores properties that don't have a `Column` attribute. It only assigns/converts properties that are returned by `ModelInstantiator#dataObjectPropertyColumns`
